File: ELP-Backend/backend/src/main/java/com/elearning/backend/entity/Course.java
package com.elearning.backend.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.util.List;

@Data
@Entity
@Table(name="courses")
public class Course {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(name = "instructor_id", nullable = false)
    private Long instructorId;
    private String domain;
    private String level;
    private Integer durationHrs;

    @ElementCollection(fetch = FetchType.EAGER)
    private List<String> tags;

    @Column(length = 1000)
    private String description;
    private String thumbnail;
    private String videoUrl;

    private String preRequisite;


}

File: E-LEARNING-PLATFORM/src/app/models/course.ts
export interface Course {
  id: number;
  title: string;
  instructorId: number;
  domain: string;
  level: string;
  durationHrs?: number;
  tags?: string;
  description?: string;
  price?: number;
  rating?: number;
  studentsCount?: number;
  thumbnail?: string;
  videoUrl?: string;
  displayTags?: string[];
  enrolled?: boolean; 
  
  //New
  avgRating?: number|null;
  preRequisite?: string;
}